### PR TITLE
Make all toImpl overloads private

### PR DESCRIPTION
**Pull request by JackStouffer**
http://forum.dlang.org/post/hapxiqrdmjdiavqlpzlv@forum.dlang.org

### CODE REVIEW

**Comment by BBasile** [comment]
after a [check](http://dtest.thecybershadow.net/artifact/website-f8ef3485d0c7d7dda9a0bc9ac55bcebca0f57e81-cfaf59aa9b8a11590df6a31fd087168c/web/phobos-prerelease/std_conv.html#.to) the documentation generated becomes totaly inconsistent:
- template to(T) , _the to function..._:  what ? function ? template ? 
- from type __Source__ to type __Target__: it's a template with a single parameter named __T__.

one solution is to write the doc in a `version(D_Ddoc){}` block, as it is already done here and there in phobos.

```D
version(D_Ddoc)
{
    /// the ddoc and the declaration used to generate the doc. 
    auto to(Source, Target)(Source source) {}
}
else
{
    // the real content
}
```

**Comment by schuetzm** [comment]
There are now lots of instances of "to" that need an "_" added, because Ddoc makes them bold otherwise.

Also, not directly related to this PR, but IMO the examples in the documentation would profit from UFCS to make them more readable.

**Comment by JackStouffer** [comment]
@BBasile @schuetzm Addressed comments

**Comment by JackStouffer** [comment]
@MartinNowak Is this breakage acceptable?

**Comment by BBasile** [comment]
Thanks for listening although I would have prefered the cheat indicates earlyer with D_Ddoc. Anyway ok now.

Another thing: there was an old enhancement request that was closed recently (handling of hex prefix in this function). Could you add a note here, after the grammar for the unsigned numbers:

```
For conversion to unsigned types, the grammar recognized is:

    UnsignedInteger:
       DecimalDigit
       DecimalDigit UnsignedInteger
```

Something like

```Note that D hexadecimal and binary literals are not handled. Neither the prefixes that indicate the base, nor the horizontal bar used to separate groups of digits are recognized. This also applies to the suffixes that indicate the type.```

**Comment by JackStouffer** [comment]
@BBasile Can you post a link to that bug so I can link to it in the docs?

**Comment by BBasile** [comment]
It's not a bug: https://issues.dlang.org/show_bug.cgi?id=9797.

It's something that won't be done, so the ddoc must specifies that certain forms of literals are not accepted.

**Comment by JackStouffer** [comment]
@BBasile I added a note with an example

File: std/conv.d [hunk 1]
```diff
@@ -306,6 +210,181 @@ template to(T)
...
+/**
+ * Conversions operate transitively, meaning that they work on arrays and
+ * associative arrays of any complexity.
+ *
+ * This conversion works because `to!short` applies _to an `int`, `to!wstring`
+ * applies _to a `string`, `to!string` applies _to a `double`, and
+ * `to!(double[])` applies _to an `int[]`. The conversion might throw an
+ * exception because `to!short` might fail the range check.
+ */
+pure unittest
```
Comment (BBasile) [inline std/conv.d]: `pure` seems to be the origin of the latest auto tester failure. Old UT for AA at line 1096 was not pure !

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by BBasile** [comment]
LGTM :heavy_check_mark:

**Comment by JackStouffer** [comment]
Please don't merge until @MartinNowak has given approval. If anyone would be against this breakage it would be him.

**Comment by BBasile** [comment]
lgtm for master. if martin doesn't pick it on stable that's another story.

**Comment by JackStouffer** [comment]
I take back my previous request. If @MartinNowak doesn't like it he can always revert it before the release.

**Comment by wilzbach** [comment]
LGTM^3 - maybe we should add a small warning to the changelog?

**Comment by JackStouffer** [comment]
@wilzbach done

**Comment by wilzbach** [comment]
> This branch has conflicts that must be resolved

I assume the changelog file?
Yet another reason for #4228 ;-)

**Comment by JackStouffer** [comment]
rebased

**Comment by JackStouffer** [comment]
ping @DmitryOlshansky I think this is good to go

**Comment by schuetzm** [comment]
There are still tons of bold "to"s:
http://dtest.thecybershadow.net/artifact/website-d6d12c1c1afc9a7a52b48c36a2c67cf74b2cf7ba-10599644e2d3a19f38cc7fdf45b92aaf/web/phobos-prerelease/std_conv.html

**Comment by JackStouffer** [comment]
@schuetzm I think I caught them all now

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**